### PR TITLE
Update donations block icon to heart gridicon

### PR DESCRIPTION
**Pull request by blackjackkent**
Fixes https://github.com/Automattic/wp-calypso/issues/44796

#### Changes proposed in this Pull Request:

* This PR updates the icon displayed when selecting a donations block in the block editor, setting it to be a Gridicons outline heart.

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Set up a test site:
    * Create a JN site running this branch.
    * Go to /wp-admin/options-general.php?page=companion_settings.
    * Activate JETPACK_BETA_BLOCKS.
    * Set JETPACK__SANDBOX_DOMAIN with your sandbox address (make sure you enter the short URL e.g. mysandbox.wordpress.com to avoid issues with the SSL certificate).
* Sandbox the API, the store and subscribe.wordpress.com (see PCYsg-lW4-p2 #sandbox-method for detailed instructions).
* Set up Jetpack and purchase a Premium or Professional plan.
* Go to Posts > New and add a donation block.
* Verify that the block icon is a heart outline.

#### Proposed changelog entry for your changes:

* Not needed.

### CODE REVIEW

File: extensions/blocks/donations/icon.js [hunk 1]
```diff
@@ -1,20 +1,6 @@
 /**
  * External dependencies
  */
-import { SVG, Path, Circle } from '@wordpress/components';
+import Gridicon from '../../../_inc/client/components/gridicon';
```
Comment (jeherve) [inline extensions/blocks/donations/icon.js:4]: This isn't going to work on WordPress.com, where that directory does not exist.

What would you think about doing something like this instead?
https://github.com/Automattic/jetpack/blob/409f77474ea16cf16691e2ee2374655d65a74e9c/extensions/shared/components/upgrade-nudge/index.jsx#L4

**Review by jeherve — CHANGES_REQUESTED** [review]

File: extensions/blocks/donations/icon.js (hunk 1 above)
Comment (blackjackkent) [reply extensions/blocks/donations/icon.js]: Makes sense! I had a feeling that might turn out not to be the ideal import statement. Thanks so much for the pointer!

**Comment by blackjackkent** [comment]
Updated per comment above! @jeherve

**Review by mmtr — APPROVED** [review]
Nice!

Screen Shot 2020-08-20 at 10 54 26
Screen Shot 2020-08-20 at 10 54 36

**Pull request merged**